File: app/src/main/java/ir/ghararemaghzha/game/fragments/MessagesFragment.java
package ir.ghararemaghzha.game.fragments;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.textview.MaterialTextView;

import io.realm.Realm;
import io.realm.RealmResults;
import io.realm.Sort;
import ir.ghararemaghzha.game.R;
import ir.ghararemaghzha.game.adapters.IncomingAdapter;
import ir.ghararemaghzha.game.models.MessageModel;

import static ir.ghararemaghzha.game.classes.Const.GHARAREHMAGHZHA_BROADCAST;


public class MessagesFragment extends Fragment {
    private Context context;
    private FragmentActivity activity;
    private Realm db;
    private MaterialTextView empty;
    private RealmResults<MessageModel> data;
    private RecyclerView recyclerView;
    private IncomingAdapter adapter;

    public MessagesFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_messages, container, false);
        context = getContext();
        activity = getActivity();
        init(v);
        return v;
    }

    private void init(View v) {
        ((MaterialTextView)activity.findViewById(R.id.toolbar_title)).setText(R.string.message_incoming);


        db = Realm.getDefaultInstance();

        db.executeTransaction(realm -> {
            RealmResults<MessageModel> results = realm.where(MessageModel.class).equalTo("sender", "admin").equalTo("read", 0).findAll();
            results.setInt("read", 1);
        });
        data = db.where(MessageModel.class).equalTo("sender", "admin").sort("date", Sort.DESCENDING).findAll();

        Intent intent = new Intent();
        intent.setAction(GHARAREHMAGHZHA_BROADCAST);
        context.sendBroadcast(intent);

        empty = v.findViewById(R.id.message_empty);

        recyclerView = v.findViewById(R.id.message_recycler);
        recyclerView.setLayoutManager(new LinearLayoutManager(context));
        if (data.isEmpty())
            empty.setVisibility(View.VISIBLE);
        else {
            adapter = new IncomingAdapter(context, data);
            recyclerView.setAdapter(adapter);
            adapter.notifyDataSetChanged();
        }


    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        if (db != null) db.close();
    }


}

File: app/src/main/java/ir/ghararemaghzha/game/fragments/MessagesFragment.kt
package ir.ghararemaghzha.game.fragments

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.navigation.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.google.android.material.textview.MaterialTextView
import io.realm.Realm
import io.realm.RealmResults
import io.realm.Sort
import io.realm.kotlin.where
import ir.ghararemaghzha.game.R
import ir.ghararemaghzha.game.adapters.IncomingAdapter
import ir.ghararemaghzha.game.classes.Const.GHARAREHMAGHZHA_BROADCAST
import ir.ghararemaghzha.game.databinding.FragmentMessagesBinding
import ir.ghararemaghzha.game.models.MessageModel

class MessagesFragment : Fragment(R.layout.fragment_messages) {
    private lateinit var b: FragmentMessagesBinding

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        b = FragmentMessagesBinding.inflate(layoutInflater, container, false)
        return b.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        init()
    }

    private fun init() {
        requireActivity().findViewById<MaterialTextView>(R.id.toolbar_title).setText(R.string.message_incoming)
        val db = Realm.getDefaultInstance()
        db.executeTransaction {
            val results: RealmResults<MessageModel> = it.where<MessageModel>().equalTo("sender", "admin").equalTo("read", "0".toInt()).findAll()
            results.setInt("read", 1)
        }
        val data = db.where<MessageModel>().equalTo("sender", "admin").sort("date", Sort.DESCENDING).findAll()
        val intent = Intent()
        intent.action = GHARAREHMAGHZHA_BROADCAST
        requireContext().sendBroadcast(intent)
        b.messageRecycler.layoutManager = LinearLayoutManager(requireContext())
        if (data.isEmpty())
            b.messageEmpty.visibility = View.VISIBLE
        else {
            val adapter = IncomingAdapter(requireContext(), requireView().findNavController(), data)
            b.messageRecycler.adapter = adapter
            adapter.notifyDataSetChanged()
        }
    }
}